src/Graphics/Effect/Effect.cs

```diff
@@ -1006,19 +1016,30 @@ IntPtr _type
 						{
 							memSize *= mem[j].info.elements;
 						}
-						memList.Add(new EffectParameter(
-							Marshal.PtrToStringAnsi(mem[j].name),
-							null,
-							(int) mem[j].info.rows,
-							(int) mem[j].info.columns,
-							(int) mem[j].info.elements,
-							XNAClass[(int) mem[j].info.parameter_class],
-							XNAType[(int) mem[j].info.parameter_type],
-							IntPtr.Zero, // FIXME: Nested structs! -flibit
-							null,
-							parameter.values + curOffset.ToInt32(),
-							memSize * 4
-						));
+						EffectParameter toAdd = new EffectParameter(
+								Marshal.PtrToStringAnsi(mem[j].name),
+								null,
+								(int) mem[j].info.rows,
+								(int) mem[j].info.columns,
+								(int) mem[j].info.elements,
+								XNAClass[(int) mem[j].info.parameter_class],
+								XNAType[(int) mem[j].info.parameter_type],
+								null, // FIXME: Nested structs! -flibit
+								null,
+								param.value.values + curOffset.ToInt32(),
```

Comment:
is there a specific reason why this changed from `parameter.values` to `param.value.values`?

Reply:
An oversight. Fixed in my working dir.